File: src/test/java/easy/_234_palindrome_linked_list/SolutionTest.java
package easy._234_palindrome_linked_list;

import org.junit.Assert;
import org.junit.Test;
import utils.ListNode;

import static org.junit.Assert.*;

public class SolutionTest {
    @Test
    public void testCase10() {

        var s = new Solution();
        var actual = s.isPalindrome(null);
        Assert.assertEquals(true, actual);
    }

    @Test
    public void testCase1() {
        var input = ListNode.getListFromArray(new int[]{1, 2});
        var s = new Solution();
        var actual = s.isPalindrome(input);
        Assert.assertEquals(false, actual);
    }

    @Test
    public void testCase2() {
        var input = ListNode.getListFromArray(new int[]{1, 2, 2, 1});
        var s = new Solution();
        var actual = s.isPalindrome(input);
        Assert.assertEquals(true, actual);
    }

    @Test
    public void testCase3() {
        var input = ListNode.getListFromArray(new int[]{1, 2, 1});
        var s = new Solution();
        var actual = s.isPalindrome(input);
        Assert.assertEquals(true, actual);
    }

    @Test
    public void testCase4() {
        var input = ListNode.getListFromArray(new int[]{1, 2, 3, 4, 2, 1});
        var s = new Solution();
        var actual = s.isPalindrome(input);
        Assert.assertEquals(false, actual);
    }

    @Test
    public void testCase5() {
        var input = ListNode.getListFromArray(new int[]{1, 2});
        var s = new Solution();
        var actual = s.reverseRecursively(input);
        Assert.assertEquals(1, 1);
    }
}

File: leetcode-python/easy/_1470_shuffle_the_array/solution_test.py
import unittest
from .solution import Solution

class MyTestCase(unittest.TestCase):

    def setUp(self) -> None:
        self.s = Solution()


    def test_case1(self):
        actual = self.s.shuffle([2,5,1,3,4,7], 3)
        self.assertEqual([2,3,5,4,1,7], actual)

    def test_case2(self):
        actual = self.s.shuffle([1,2,3,4,4,3,2,1], 4)
        self.assertEqual([1,4,2,3,3,2,4,1], actual)

    def test_case3(self):
        actual = self.s.shuffle([1,1,2,2], 2)
        self.assertEqual([1,2,1,2], actual)


if __name__ == '__main__':
    unittest.main()
